pkg/buildscript/unmarshal_buildexpression.go

```diff
@@ -172,7 +175,7 @@ func unmarshalValue(path []string, valueInterface interface{}) (*value, error) {
 		result.List = &values
 
 	case string:
-		if sliceutils.Contains(path, ctxIn) || strings.HasPrefix(v, "$") {
+		if len(path) >= 2 && path[len(path)-2] == ctxIn || strings.HasPrefix(v, "$") {
```

Comment:
😬 We should probably prioritize the story to revisit the `path` value.

Reply:
I'll clean it up a little. At the moment no story is planned, since Mitchell said it's otherwise all working as intended, this is a one off bug.

That said we will surely be refactoring this library as a whole once buildscripts are more stable.